app/src/main/java/com/arctouch/codechallenge/view/MovieDetailFragment.kt

```diff
@@ -0,0 +1,45 @@
+package com.arctouch.codechallenge.view
+
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder
+import com.arctouch.codechallenge.viewmodel.NavigationViewModel
+import kotlinx.android.synthetic.main.movie_detail_fragment.*
+
+class MovieDetailFragment : Fragment() {
+    //viewmodel
+    private lateinit var navigationViewModel: NavigationViewModel
+
+    companion object {
+        fun newInstance() = MovieDetailFragment()
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return inflater.inflate(R.layout.movie_detail_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        //initialize viewmodel
+        navigationViewModel = ViewModelProviders.of(activity!!).get(NavigationViewModel::class.java)
+
+        val movie = navigationViewModel.getMovie()
```

Comment:
Hello Danilo!
That's the easly form that i found with the time. But you can pass this object on the fragment instance, or, in my studies i found a solution using rxjava. I'll take a look in another ways.

Reply:
Could elaborate more on your fragment instance idea? How would you pass it from one fragment to the other?

Reply:
What I thought was to use the same method of passing the data through the viewmodel but send it in the fragment instance method instead of a get() method inside the viewmodel.

Did you have another best way to do that?

Reply:
> fragment instance method
Do you mean, `newInstance()`? And what about when the fragment is recreated because of a orientation change?

Reply:
The data can be saved in the SavedInstanceState as a parcelable object.